units/include/scipp/units/unit.tcc

```diff
@@ -29,28 +31,50 @@ template <class T> std::string to_string(const T &unit) {
 }
 } // namespace units
 
-template <class T> std::string Unit_impl<T>::name() const {
+template <class T, class Counts>
```

Comment:
I see `Counts` is a special case in all the function template declarations here. Could there be other ones in the future that also need to be listed explicitely, potentially making this quite cumbersome?
```
template <class T, class Counts, class OtherCounts, class AThirdOne>
std::string Unit_impl<T, Counts, OtherCounts, AThirdOne>::name() const {
```
And you would have to remember the order.
Maybe `Counts` is the only one. Can't really think of others right now.

Reply:
Me neither. There is things like angles which may require special units, but they are not linked into so many operations' behaviors, i.e., we may not need special support.